File: Sources/gestion-culturelle-back/src/main/java/com/afpa/cda/dto/ManifestationDto.java
package com.afpa.cda.dto;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class ManifestationDto {
	private int id;
	private String label;
	private Date dateValidation;
	private AdminDto validateur;
	private AnimationDto animation;
	private Date dateDebut;
	private Date dateFin;
	private double cout;
	private SalleDto salle;
	private double prixBillet;
	private int reservations;
	private int reservationsVip;
	private int rentabilite;
	private AdminDto annulateur;
	private Date dateAnnulation;

	private List <CommandeDto> listCommandes;

	private List <UserDto> listVips;
}

File: Sources/gestion-culturelle-front/project/src/app/model/manifestationDto.ts
import { SalleDto } from "./salleDto";
import { AnimationDto } from './animationDto';
import { AdminDto } from "./adminDto";
import { CommandeDto } from "./commandeDto";
import { User } from './user';

export class ManifestationDto {
    id: number;
    label: string;
    dateValidation: Date;
    validateur: AdminDto;
    animation: AnimationDto;
    dateDebut: Date;
    dateFin: Date;
    cout: number;
    salle: SalleDto;
    prixBillet: number;
    reservations: number;
    reservationsVip: number;
    rentabilte: number;
    annulateur: AdminDto;
    dateAnnulation: Date;
    listCommandes : CommandeDto;
    listVips : User;


    constructor(id?: number,
        label?: string,
        dateValidation?: Date,
        validateur?: AdminDto,
        animation?: AnimationDto,
        dateDebut?: Date,
        dateFin?: Date,
        cout?: number,
        reservations?: number,
        reservationsVip?: number,
        rentabilte?: number,
        prixBillet?: number,
        salle?: SalleDto,
        annulateur?: AdminDto,
        dateAnnulation?: Date,
        listCommandes ?:CommandeDto,
        listVips?: User,
    ) {

        this.id = id;
        this.label = label;
        this.dateValidation = dateValidation;
        this.validateur = validateur;
        this.animation = animation;
        this.dateDebut = dateDebut;
        this.dateFin = dateFin;
        this.cout = cout;
        this.prixBillet = prixBillet;
        this.reservations = reservations;
        this.reservationsVip = reservationsVip;
        this.rentabilte = rentabilte;
        this.salle = salle;
        this.annulateur = annulateur;
        this.dateAnnulation = dateAnnulation;
        this.listCommandes = listCommandes;
        this.listVips = listVips;

    }
}


